### ISSUE TITLE
Linphone doesn't stay connected over OpenVPN

### ISSUE THREAD

**Issue by afriedmanGlacier** [BelledonneCommunications/linphone-iphone#129]
We have been using Linphone over OpenVPN Connect on iOS for our project and have a requirement for the VPN to stay connected and Linphone available even while sleeping. Unfortunately, the VPN disconnects as Apple has strict parameters that only allow a specific VPN setup (IKEv2 with other caveats) to stay connected. However, we also tested with Bria VOIP and noticed that the VPN seems to stay connected while a user is registered and enabled in Bria. I would LOVE for this to be the same with Linphone because we'd much rather use an open source solution for VOIP.

Apple does allow VOIP apps to stay running in the background, and I assume Linphone has this as a setting, but I can't figure out what else might allow Bria to keep the VPN connected, but it doesn't stay connected with Linphone. Another thought I had is that it has something to do with the recent CallKit integration (I have not tested old versions of Bria without that integration)

Again, we'd rather use open source VOIP and Linphone. Any ideas how something like Bria keeps the OpenVPN connected and whether that can be done with Linphone?

**Comment by eliburke** [comment]
I have also noticed that the new Bria app is able to stay connected in the background, unlike our own Linphone-derived app where this behavior "broke" after the upgrade to iOS10. They are using the CallKit UI, so they must be building with Xcode 8. 

My personal speculation is that it has to do with PushKit, aka VOIP Push notifications. I think that Apple still allows active background connections for VOIP apps, unless you register for high priority VOIP notifications.

**Comment by JFSene** [comment]
Apple deprecated every background connection, you have to use Voip Pushes in order to wake the app while it is in its idle stage. An CallKit works beautifully with VoipP Push, the problem is with Linphone's submodules and codecs, it makes using CallKit almost impossible.

**Comment by afriedmanGlacier** [comment]
I heard back from a developer at a competitor to Bria and he said their app stays awake in the background (thereby keeping my OpenVPN connection up) by playing a silent audio file periodically. I guess they can get away with that and still be approved for the app store because it IS a VoIP app and can legitimately set the VoIP and audio background modes. Not sure if this makes sense as a default setting for Linphone (it drains battery quite a bit), but perhaps as an optional switch. I was able to get this to work by setting AudioSession settings and starting an AVPlayer on applicationDidEnterBackground with AVPlayerActionAtItemEndNone and then pausing it on applicationDidBecomeActive. (I followed the sample at https://yifan.lu/2013/12/17/unlimited-backgrounding-on-ios/)
